### ISSUE TITLE
[FEATURE] - Lidarr/Readarr - Soularr/slskd support

### ISSUE THREAD

**Issue by OCT0PUSCRIME** [RandomNinjaAtk/arr-scripts#320]
**Is your feature request related to a problem? Please describe.**

Finding niche tracks

**Describe the solution you'd like**
Looking for slskd integration to use soulseek to import niche tracks that can't be found with trackers or arr-scripts

**Describe alternatives you've considered**
Soularr - see below

**Additional context**

Apologies if this is the wrong forum. I thought you used to have a discord, but I can't find it.

There is a new project [Soularr](https://github.com/mrusse/soularr) (and it's fork with readarr support) that I am interested in trying out. I currently use slskd to manually download niche tracks that my trackers and the arr-scripts can't find.

Do you see any conflicts with deploying this alongside the arr-scripts?

I marked this as a feature request in the event that you might be interested in implementing a similar solution.

**Comment by SickProdigy** [comment]
I run slsk and just seen it too. Would be nice to just add another option and few config settings than downloading another separate instance to run a script in.

**Comment by OCT0PUSCRIME** [comment]
> I run slsk and just seen it too. Would be nice to just add another option and few config settings than downloading another separate instance to run a script in.

Agreed. If possible it would be cool. If not I still would like some input on if it might not play well with the arr-scripts.

**Comment by RandomNinjaAtk** [comment]
Interesting, first I've heard of this project... No promises might look into it when I have the chance.

**Comment by hockeygoalie35** [comment]
I'm taking a look into this. The soularr project is great, but it doesn't have any of the beets tagging/ path naming / audio processing. Also ARL tokens are sort of going the way of the dodo, as Deezer is cracking down on ARLs.

**Comment by dnick5000** [comment]
I've been using both. The only issue I've seen is one will try to download an album that the other one has already gotten. Otherwise, Soularr has been a great supplement to what Arr-script can't find.

**Comment by hockeygoalie35** [comment]
I've add a pull request to Soularr #https://github.com/mrusse/soularr/pull/86, to allow the functions be used independently. It'd be nice to be able to write a script that can be called by the Audio service, and then be tagged by beets and Lidarr/Plex notified.

**Comment by hockeygoalie35** [comment]
Just a small update, working on a proof of concept that hooks Audio script to Slskd. I'm using a lot of the same logic as Soularr, but my script is tailor-made to accept Audioscript's input.


### Script output:
![Image](https://github.com/user-attachments/assets/eead23e8-4c6c-404d-a4ac-fcf2ab70b146)

### Slskd Search:

![Image](https://github.com/user-attachments/assets/6060e72f-c39e-48e5-8cc8-d07be4a886b3)

**Comment by OCT0PUSCRIME** [comment]
> Just a small update, working on a proof of concept that hooks Audio script to Slskd. I'm using a lot of the same logic as Soularr, but my script is tailor-made to accept Audioscript's input.
> 
> ### Script output:
> ![Image](https://github.com/user-attachments/assets/eead23e8-4c6c-404d-a4ac-fcf2ab70b146)
> 
> ### Slskd Search:
> ![Image](https://github.com/user-attachments/assets/6060e72f-c39e-48e5-8cc8-d07be4a886b3)

Awesome thanks for the update. Happy to test!

**Comment by OCT0PUSCRIME** [comment]
The lidarr plugins branch now has a [slskd plugin](https://wiki.servarr.com/en/lidarr/plugins#slskd) that I am planning on testing.
